src/sagemaker/workflow/steps.py

```diff
@@ -301,6 +301,7 @@ def arguments(self) -> RequestType:
         )
         request_dict = self.estimator.sagemaker_session._get_train_request(**train_args)
         request_dict.pop("TrainingJobName")
+        request_dict["HyperParameters"].pop("sagemaker_job_name", None)
```

Comment:
wonder if it is possible that this request dict does not have `HyperParameters` in it?

Reply:
It will always be there, but let me add a safety check